programs/fileio.c

```diff
@@ -770,27 +766,73 @@ static size_t FIO_createDictBufferMMap(void** bufferPtr, const char* fileName, F
         }
     }
 
-    *bufferPtr = FIO_mmap((size_t)fileSize, fileHandle);
+    *bufferPtr = mmap(NULL, (size_t)fileSize, PROT_READ, MAP_PRIVATE, fileHandle, 0);
```

Comment:
minor : check that `fileSize` is not `> (U64)SIZE_MAX`, it's a prerequisite for safe casting to `(size_t)`.

(note: this situation could happen for files > 4 GB on 32-bit systems).